server/models/Order.ts

```diff
@@ -1,312 +1,478 @@
 import { TaxType } from '@opencollective/taxes';
 import debugLib from 'debug';
 import { get } from 'lodash';
-import {
-  ForeignKey,
-  HasManyGetAssociationsMixin,
-  HasOneGetAssociationMixin,
-  InferAttributes,
-  InferCreationAttributes,
-  Model,
-  ModelStatic,
-} from 'sequelize';
+import * as Sequelize from 'sequelize';
+import { DataTypes, Model, Op, Optional, QueryTypes } from 'sequelize';
 import Temporal from 'sequelize-temporal';
 
-import { roles } from '../constants';
 import OrderStatus from '../constants/order-status';
 import { PAYMENT_METHOD_SERVICE, PAYMENT_METHOD_TYPE } from '../constants/paymentMethods';
+import roles from '../constants/roles';
 import TierType from '../constants/tiers';
 import { PLATFORM_TIP_TRANSACTION_PROPERTIES, TransactionTypes } from '../constants/transactions';
 import * as libPayments from '../lib/payments';
-import sequelize, { DataTypes, Op, QueryTypes } from '../lib/sequelize';
+import sequelize from '../lib/sequelize';
 import { sanitizeTags, validateTags } from '../lib/tags';
 import { capitalize } from '../lib/utils';
 
-import AccountingCategory from './AccountingCategory';
+import type AccountingCategory from './AccountingCategory';
+import type Activity from './Activity';
 import Collective from './Collective';
-import CustomDataTypes from './DataTypes';
-import { MemberModelInterface } from './Member';
+import type Comment from './Comment';
+import type { MemberModelInterface } from './Member';
 import PaymentMethod, { PaymentMethodModelInterface } from './PaymentMethod';
-import { SubscriptionInterface } from './Subscription';
-import Tier from './Tier';
-import { TransactionInterface } from './Transaction';
+import type { SubscriptionInterface } from './Subscription';
+import type Tier from './Tier';
+import Transaction, { TransactionInterface } from './Transaction';
 import User from './User';
 
 const { models } = sequelize;
 
 const debug = debugLib('models:Order');
 
-interface OrderModelStaticInterface {
-  generateDescription(collective, amount, interval, tier): string;
-  cancelActiveOrdersByCollective(collectiveId: number): Promise<[affectedCount: number]>;
-  cancelActiveOrdersByTierId(tierId: number): Promise<[affectedCount: number]>;
-  cancelNonTransferableActiveOrdersByCollectiveId(collectiveId: number): Promise<[affectedCount: number]>;
+export interface OrderAttributes {
+  id: number;
+  CreatedByUserId?: number;
+  CollectiveId?: number;
+  currency?: string;
```

Comment:
We'll want to update this one for https://github.com/opencollective/opencollective-api/pull/9706

Reply:
todo